kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -544,6 +606,27 @@ private static <T extends HasMetadata> boolean isStatusFresh(T resource, Functio
         }
     }
 
+    private static <T extends CustomResource<?, ?>> ResourceCheckResult<T> handleListener(T resource, StrimziKafkaRef strimziKafkaRef,
+                                                                                          StatusFactory<T> statusFactory,
+                                                                                          Kafka kafka) {
+        if (!isListenerPresent(strimziKafkaRef, kafka)) {
+            return new ResourceCheckResult<>(statusFactory.newFalseConditionStatusPatch(resource, ResolvedRefs,
```

Comment:
So I think there are two conditions here, not one.

If the listener is not present in the **spec**, then it means the Kroxylicious has misconfigured their KafkaServiceCR.  This is a permanent condition and something the user needs to fix.

"Referenced resource does not declare listener name: blancmange"
Condition.REASON_INVALID_REFERENCED_RESOURCE

If the listener is not present in the **status**, then it probably mean that the Strimzi operator is stil reconciling.  This might be a transient condition that will clear once Strimzi Operator does its thing.

"Referenced resource has not yet reconciled listener name: blancmange"
Condition.REASON_REFERENCED_RESOURCE_NOT_RECONCILED